sdk/src/Core/Amazon.Runtime/Credentials/Internal/IdentityResolvers/IdentityResolverConfiguration.cs

```diff
@@ -23,17 +23,20 @@ namespace Amazon.Runtime.Credentials.Internal.IdentityResolvers
     public class DefaultIdentityResolverConfiguration : IIdentityResolverConfiguration
     {
         public static readonly IIdentityResolverConfiguration Instance = new DefaultIdentityResolverConfiguration();
-        private static readonly Dictionary<Type, IIdentityResolver> identityResolvers = new Dictionary<Type, IIdentityResolver>()
-            {
-                { typeof(AnonymousAWSCredentials), new AnonymousIdentityResolver() },
-                { typeof(AWSCredentials), new DefaultAWSCredentialsIdentityResolver() },
-            };
+        private static readonly Dictionary<Type, IIdentityResolver> identityResolvers = new()
+        {
+            { typeof(AnonymousAWSCredentials), new AnonymousIdentityResolver() },
+            { typeof(AWSCredentials), new DefaultAWSCredentialsIdentityResolver() },
+            { typeof(AWSToken), new DefaultAWSTokenIdentityResolver() },
+        };
 
         /// <inheritdoc/>
         public IIdentityResolver GetIdentityResolver<T>() where T : BaseIdentity
         {
-            if(identityResolvers.TryGetValue(typeof(T), out var identityResolver))
+            if (identityResolvers.TryGetValue(typeof(T), out var identityResolver))
```

Comment:
Seeing this `typeof` calls and working with type properties reminds we should a NativeAOT compilation test. I don't want to rely only on the warnings as that has burned us in the past.

Reply:
I asked about this on a previous PR: https://github.com/aws/aws-sdk-net/pull/3531#discussion_r1823088083

Reply:
I'm pretty sure you are good with NativeAOT because you aren't asking for anything that won't be known at compile time. I just think as a sanity check at the end of SRA when we are doing bug bashing we should include some NativeAOT testing.